tests: power: Fix compiler definition flag

mnirmaladevi opened the pull request:
Enabled compiler definition flag TEST_CASE_SLEEP_SUCCESS
which was missing on migrating to cmake.

Signed-off-by: Nirmala Devi <nirmala.devix.m@intel.com>

nashif reviewed (changes requested):
docs for the sample need to be updated, still talking about 

make TEST_CASE=sleep-fail

mnirmaladevi committed:
tests: power: Fix compiler definition flag

Enabled compiler definition flag TEST_CASE_SLEEP_SUCCESS
which was missing on migrating to cmake.

Signed-off-by: Nirmala Devi <nirmala.devix.m@intel.com>
